### PR TITLE
Change PERSIST_FILE_WATERMARK size

### PR DESCRIPTION
**Pull request by szemere**
Persist file uses a "pre-allocation" mechanism to circumvent some side effects of a running out of space scenario.

More on that in PR #1785, especially in 45a4157.

Thus PERSIST_FILE_WATERMARK must be bigger than any entry we want to store in the persist file. (Including the extra headers and flags.)

One of our internal module has an 8k buffer. Considering the extra space for the headers, the current value is: 8k+256 bytes.

---
Note: I have two alternative follow up plans. Currently work in progress, but I do not want to block this fix: 
A) https://github.com/szemere/syslog-ng/tree/persist_changes_wip Create a static assert on struct sizes. 
B) https://github.com/szemere/syslog-ng/tree/persist_changes_wip2 Eliminate the pre-allocation logic. This way we can always check on the real alloc size, so there is no need for heuristics.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by MrAnno** [comment]
@kira-syslogng do stresstest

**Comment by kira-syslogng** [comment]
Kira-stress-test: Build SUCCESS

**Review by MrAnno — APPROVED** [review]
https://github.com/szemere/syslog-ng/tree/persist_changes_wip2 seems to be a good idea, it would be a nice improvement.

File: lib/persist-state.c [hunk 1]
```diff
@@ -37,7 +37,7 @@
 
 #define PERSIST_FILE_INITIAL_SIZE 16384
 #define PERSIST_STATE_KEY_BLOCK_SIZE 4096
-#define PERSIST_FILE_WATERMARK 4096
+#define PERSIST_FILE_WATERMARK 8448
```
Comment (bazsi) [inline lib/persist-state.c:40]: this should be aligned on a 4k boundary.

Comment (bazsi) [reply lib/persist-state.c:40]: let's add an assert into the persist file allocation function that would limit the allocation to sizes that we can actually guarantee.

**Review by bazsi — DISMISSED** [review]

File: lib/persist-state.c (hunk 1 above)
Comment (MrAnno) [reply lib/persist-state.c:40]: Is the 4k page-boundary necessary? `PERSIST_FILE_WATERMARK` seems to be just a free-space threshold, syslog-ng always allocates `PERSIST_FILE_INITIAL_SIZE=16384` chunks, and `_grow_store()` makes sure it's page-aligned.

Comment (MrAnno) [reply lib/persist-state.c:40]: Is the 4k page-boundary necessary? `PERSIST_FILE_WATERMARK` seems to be just a free-space threshold, syslog-ng always allocates `PERSIST_FILE_INITIAL_SIZE=16384` chunks, and `_grow_store()` makes sure it's page-aligned with `getpagesize()` (I think that might return 8 and 16k as well).

Comment (bazsi) [reply lib/persist-state.c]: Ok, that's ok this way then.

Comment (MrAnno) [reply lib/persist-state.c]: @bazsi Thank you.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by MrAnno — APPROVED** [review]

**Review by bazsi — APPROVED** [review]

**Pull request merged**